### DISCUSSION TITLE
How to get request in ModelView?

### DISCUSSION THREAD

**Discussion by KNMohamed** [aminalaee/sqladmin#729] [Q&A]
Hello, 
I have a class that inherits from the ModelView class and uses a **custom wtform** with a **SelectMultipleField**. 
This SelectMultipleField is a relation to another Model, however I need to filter out the choices that are shown in this model based on an attribute I have set in the **request.session**. How can I achieve this?

**Comment by aminalaee** [comment]
Hey, in which method do you exactly need to access the request object?

**Reply by KNMohamed** [reply]
I need to access it at the class level. I am creating a custom form by overriding the form attribute
https://aminalaee.dev/sqladmin/configurations/#form-options

The reason being, is I need to filter out the relationships in the edit form based on a property set on request.session in a different custom view. I am trying to achieve this by creating a custom form and setting the choices at runtime. 

Here is a similar discussion where a user is trying to filter out relationships in the edit form. 
https://github.com/aminalaee/sqladmin/discussions/712

**Comment by lesleslie** [comment]
I don't have an example, but how about copying the request object, either in the route or in middleware, to a contextvar? Then it can be accessed from any function at the class level.

**Reply by lesleslie** [reply]
I just came across this example from starlette-core: https://github.com/accent-starlette/starlette-core/blob/master/starlette_core/middleware.py

```
from contextvars import ContextVar

from starlette.types import ASGIApp, Receive, Scope, Send


_request_ctx_var: ContextVar[Scope] = ContextVar(
    "request", default=None  # type: ignore
)


def get_request() -> Scope:
    return _request_ctx_var.get()


class CurrentRequestMiddleware:
    """
    Sets the _request_ctx_var to the request.

    Usage:
        from starlette_core.middleware import get_request
        get_request()
    """

    def __init__(self, app: ASGIApp) -> None:
        self.app = app

    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
        if scope["type"] not in ("http", "websocket"):
            await self.app(scope, receive, send)
            return

        local_scope = _request_ctx_var.set(scope)

        response = await self.app(scope, receive, send)

        _request_ctx_var.reset(local_scope)

        return response

```

**Comment by umarr-devv** [comment]
I also ran into this problem.
It was necessary to show only certain values for SelectMultipleField and SelectField in the form, but the necessary data were in the query, which were entered there during authorization

(I used a translator, I apologize if it's unclear.)

**Reply by aminalaee** [reply]
@umarr-devv In which method do you exactly need the request?
